cloud/amazon/iam.py

```diff
@@ -279,11 +259,31 @@ def update_user(module, iam, name, new_name, new_path, key_state, key_count, key
                 else:
                     module.fail_json(msg=error_msg)
 
+    try:
+        current_keys, status = \
```

Comment:
Would it be possible to un-nest this? Nested list comprehensions can be hard to read. Also the use of `\` for line breaks doesn't conform to Ansible's PEP8-ish style guidelines.